server/src/server.ts

```diff
@@ -126,7 +126,16 @@ connection.onDidChangeWatchedFiles(({ changes }) => {
         return
       }
       const [bundleChanged, bundleValidator] = expect(workspaceBookBundles.get(workspaceChanged.uri), 'already returned if key missing')
-      bundleChanged.processChange(change)
+      if (bundleChanged.isDirectoryDeletion(change)) {
```

Comment:
Thoughts on moving this branching into the `processChange` function as a guard, so we don't have to meet "the change uri is not a directory" as an invariant for calling it and getting correct behavior? maybe it makes no difference if this is the only call site...

Reply:
Sure, we can do that.

Reply:
[Done](https://github.com/openstax/poet/pull/60/commits/8855d89caf6bd40fddbd998e917271bbba6e0dbf)